pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/JWTRouteFilterTest.java

```diff
@@ -216,4 +238,50 @@ private String createTokenWithRoles(String secret, long invalidAfter, String rol
         }
     }
 
+    public static DecodedJWT verifyDDPToken(String jwt, JwkProvider jwkProvider) {
+        DecodedJWT validToken;
+        RSAKeyProvider keyProvider;
+        try {
+            keyProvider = RSAKeyProviderFactory.createRSAKeyProviderWithPrivateKeyOnly(jwkProvider);
+        }
+        catch (Exception e) {
+            logger.error("Error creating RSAKeyProvider", e);
+            throw (e);
+        }
+
+        try {
+            validToken = JWT.require(Algorithm.RSA256(keyProvider)).acceptLeeway(10).build().verify(jwt);
+        }
+        catch (TokenExpiredException e) {
+            // TokenExpired is one of the benign variants of JWTVerificationException that the `verify()` method throws.
+            logger.warn("Expired token: {}", jwt);
+            throw e;
+        }
+        catch (Exception e) {
+            logger.error("Could not verify token {}", jwt, e);
+            throw (e);
+        }
+        return validToken;
+    }
+
+    @Test
+    public void test() {
+        Map<String, JwkProvider> jwkProviderMap = new HashMap();
+        String jwt = "<COPY_TOKEN_HERE>";
+        // We need to get the Auth0ClientId *before* verification so that we can verify see: Chicken/Egg
+        DecodedJWT decodedJwt = JWT.decode(jwt);
+        String auth0ClientId = decodedJwt.getClaim(clientId).asString();
+        String auth0Domain = decodedJwt.getClaim(tenantDomain).asString();
+        JwkProvider jwkProvider = jwkProviderMap.getOrDefault(auth0ClientId, null);
+        if (jwkProvider == null) {
+
+            jwkProvider = new JwkProviderBuilder(auth0Domain).cached(100, 3L, MINUTES).build();
+            jwkProviderMap.put(auth0ClientId, jwkProvider);
+        }
+
+        DecodedJWT validToken = verifyDDPToken(jwt, jwkProvider);
+        String ddpUserGuid = validToken.getClaim(userId).asString();
+        Assert.assertNotNull(ddpUserGuid);
+    }
+
```

Comment:
I copied the dss part of token verification here and cleared the unnecessary parts (user info, etc) to test our token with that verification method